src/edu/psu/sweng500/team8/play/ActionManager.java

```diff
@@ -36,12 +36,13 @@ public void doUndo(CellGrid currentCellGridFromBoard) {
 		 * revert last action on the sudokuActionQueue, and put that action into
 		 * sudokuActionQueueForUndo
 		 */
-		SudokuAction lastAction = sudokuActionQueue.pop();
 		
-		if (null != lastAction) {
-			CellGrid previousCellGrid = lastAction.getCellGrid();
-			lastAction.setCellGrid(new CellGrid(currentCellGridFromBoard));
-			sudokuActionQueueForUndo.add(lastAction);
+		if (null != sudokuActionQueue.peek()) {

```

Comment:
I find it very surprising that isEmpty didn't work...the only way it wouldn't is if you pushed null into the stack, in which case you have a bug.